src/Microsoft.AspNetCore.OData/Batch/ODataBatchReaderExtensions.cs

```diff
@@ -138,6 +138,9 @@ public static class ODataBatchReaderExtensions
                 request.Body = bufferedStream;
             }
 
+            // remove the content length and content type headers in case the batchRequest item doesn't have a body
+            request.ContentLength = null;
+            request.ContentType = null;
```

Comment:
Rather than set these in CreateHttpContext and then remove them here, if we don't think certain headers should be copied (and I certainly agree for content-length) then we should put the right logic in CreateHttpContext for copying the relevant headers and omitting others.  [http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_HeaderFields](http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_HeaderFields) spells out how well-known headers on the overall batch should be applied to individual statements within the batch. For headers in this list, if it doesn't specifically say that they are inherited, then we shouldn't copy. We should continue to copy headers not in this list.